src/transformStateWithClones.js

```diff
@@ -65,7 +65,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const array = [];
+  const temp = Object.assign({}, state);
+
+  for (let i = 0; i < transforms.length; i++) {
+    if (transforms[i].operation === 'addProperties') {
+      Object.assign(temp, transforms[i].properties);
+      array.push({ ...temp });
+    }
+
+    if (transforms[i].operation === 'removeProperties') {
+      for (let k = 0; k < (transforms[i].properties).length; k++) {
+        delete temp[transforms[i].properties[k]];
+      }
+      array.push({ ...temp });
```

Comment:
```suggestion
      array.push({ ...temp });
```

You do this 3 time. you can push changed state at the end of each loop